### PR TITLE
OS

### PR DESCRIPTION
**Pull request by Andy-Python-Programmer**
Added os commands
1. system (perform a system operation)
2. cwd (get the cwd())

### CODE REVIEW

**Comment by Andy-Python-Programmer** [comment]
@AkramIzz Note the test will fail as its you know a ci and where dart even in cmd will not work

File: lib/external/io.dart [hunk 1]
```diff
@@ -23,11 +25,43 @@ ASTNode funcFree(Runtime runtime, ASTNode self, List<ASTNode> args) {
...
   });
 
   return INITIALIZED_NOOP;
-}
\ No newline at end of file
+}
+
+StringNode funcCwd(Runtime runtime, ASTNode self, List<ASTNode> args) {
+  StringNode res = StringNode();
+  Directory current = Directory.current;
```
Comment (Calamity210) [inline lib/external/io.dart:39]: This line is redundant. If current is only used once, i wouldn't say its worth making a separate variable for it. Use Directory.current directly

File: lib/external/io.dart [hunk 2]
```diff
@@ -23,11 +25,43 @@ ASTNode funcFree(Runtime runtime, ASTNode self, List<ASTNode> args) {
...
+
+  res.stringValue = current.path.toString();
```
Comment (Calamity210) [inline lib/external/io.dart:41]: Return the absolute path

**Comment by Andy-Python-Programmer** [comment]
Test are failing as of echo :(

File: lib/external/io.dart [hunk 3]
```diff
@@ -9,6 +9,8 @@ import 'package:Birb/utils/scope.dart';
 void registerIO(Runtime runtime) {
   registerGlobalFunction(runtime, 'exit', funcExit);
   registerGlobalFunction(runtime, 'free', funcFree);
+  registerGlobalFunction(runtime, 'cwd', funcCwd);
```
Comment (Calamity210) [inline lib/external/io.dart:12]: This would be better to be registered as a variable than a function

**Review by Calamity210 — CHANGES_REQUESTED** [review]

File: lib/external/io.dart [hunk 4]
```diff
@@ -23,11 +25,44 @@ ASTNode funcFree(Runtime runtime, ASTNode self, List<ASTNode> args) {
...
+    'errorCode': null,
+    'stdout': null,
+    'stderr': null
+  };
+
+  MapNode res = MapNode();
+
+  await Process.run(args[0].stringValue,
+          args[1].listElements.map((e) => e.toString()).toList())
+      .then((ProcessResult pr) {
```
[the remark this answers is not in the document]
Comment (Andy-Python-Programmer) [reply lib/external/io.dart]: Done!

**Comment by Andy-Python-Programmer** [comment]
The os is destoryed as of git. Git is stressfull

**Pull request closed without merging**